smartsim/_core/mli/workermanager.py

```diff
@@ -0,0 +1,976 @@
+        """Send a message throuh the underlying communication channel
+        :param value: The value to send"""
+        self._channel.send_bytes(value)
+
+    @property
+    def descriptor(self) -> bytes:
+        """Return the channel descriptor for the underlying dragon channel"""
+        if isinstance(self._descriptor, str):
+            return self._descriptor.encode("utf-8")
+        return self._descriptor
+
+    # @property
+    # def key(self) -> str:
+    #     """Return the channel descriptor for the underlying dragon channel"""
+    #     descriptor = du.B64.bytes_to_str(self._channel.serialize())
+    #     return descriptor
+
+    @classmethod
+    def find(cls, key: bytes) -> "CommChannel":
+        """Find a channel given its serialized key
+        :param key: The unique descriptor of a communications channel"""
+        # todo: load channel correctly using dragon
+        # ch = dch.Channel(key)
+        dc = DragonCommChannel(key)
+        dc._descriptor = key
+        return dc
+
+
+class InferenceRequest:
+    """Temporary model of an inference request"""
+
+    def __init__(
+        self,
+        model_key: t.Optional[str] = None,
+        callback: t.Optional[CommChannel] = None,
+        raw_inputs: t.Optional[t.List[bytes]] = None,
+        input_keys: t.Optional[t.List[str]] = None,
+        output_keys: t.Optional[t.List[str]] = None,
+        raw_model: t.Optional[bytes] = None,
+        batch_size: int = 0,
+        device: t.Optional[str] = None,
+    ):
+        """Initialize the InferenceRequest"""
+        self.model_key = model_key
+        self.raw_model = raw_model
+        self.callback = callback
+        self.raw_inputs = raw_inputs
+        self.input_keys = input_keys or []
+        self.output_keys = output_keys or []
+        self.batch_size = batch_size
+        self.device = device
+
+    @staticmethod
+    def from_dict(source: t.Dict[str, t.Any]) -> "InferenceRequest":
+        return InferenceRequest(
+            # **source
+        )
+
+
+class InferenceReply:
+    """Temporary model of a reply to an inference request"""
+
+    def __init__(
+        self,
+        outputs: t.Optional[t.Collection[t.Any]] = None,
+        output_keys: t.Optional[t.Collection[str]] = None,
+        failed: bool = False,
+    ) -> None:
+        """Initialize the InferenceReply"""
+        self.outputs: t.Collection[t.Any] = outputs or []
+        self.output_keys: t.Collection[t.Optional[str]] = output_keys or []
+        self.failed = failed
+
+
+class ModelLoadResult:
+    """A wrapper around a loaded model"""
+
+    def __init__(self, model: t.Any) -> None:
+        """Initialize the ModelLoadResult"""
+        self.model = model
+
+
+class InputTransformResult:
+    """A wrapper around a transformed input"""
+
+    def __init__(self, result: t.Any) -> None:
+        """Initialize the InputTransformResult"""
+        self.transformed = result
+
+
+class ExecuteResult:
```

Comment:
Using the same result everywhere could work but I thought it would be extremely confusing to a worker author that wanted to be sure they were getting what they expect.

I was also hedging my bets on the idea that as we go, we would find that we need more things in the results. It's happened only once so far.

Reply:
Added 713 for follow-up task